src/test/countries.test.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import { BrowserRouter as Router } from 'react-router-dom'; // Import the BrowserRouter component
+import Countries from '../components/Countries';
+
+const mockStore = configureStore([]);
+
+test('renders the Countries component correctly', () => {
+  const initialState = {
+    allHolidays: {
+      AllHolidays: [
+        {
+          countryCode: 'US',
+          name: 'United States',
+        },
+        {
+          countryCode: 'CA',
+          name: 'Canada',
+        },
+      ],
+      isLoading: false,
+    },
+  };
+
+  const store = mockStore(initialState);
+
+  const { container } = render(
+    <Provider store={store}>
+      <Router>
+        {' '}
+        {/* Wrap the component with the Router */}
+        <Countries />
+      </Router>
+    </Provider>,
+  );
+
+  expect(container).toMatchSnapshot();
+});
+
+test('renders the loading state correctly', () => {
+  const initialState = {
+    allHolidays: {
+      AllHolidays: [],
+      isLoading: true,
+    },
+  };
+
+  const store = mockStore(initialState);
+
+  const { container } = render(
+    <Provider store={store}>
+      <Router>
+        {' '}
+        {/* Wrap the component with the Router */}
+        <Countries />
+      </Router>
+    </Provider>,
+  );
+
+  expect(container).toMatchSnapshot();
```

Comment:
- [ ]  It is recommended to have unit tests for pure functions in your project as well. Write tests for your project help it to be more stable, and to future know what went wrong in case  it crushed.

Reply:
Hey @ishimwezachee.In the video I mentioned the coolest/most interesting bit of the code I enjoyed working on in the 1.54 minute. Kindly check it out.